ODataSession.cs

```diff
@@ -0,0 +1,89 @@
+﻿using Andculture.Sitefinity.Testing.Models.Configuration;
+using AndcultureCode.CSharp.Core.Utilities.Configuration;
+using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json;
+using RestSharp;
+using System;
+using System.IO;
+
+namespace Andculture.Sitefinity.Testing
+{
+    public class ODataSession
+    {
+        #region Member Variables
+
+        private string EnvironmentName = "Testing";
+
+        private static IConfigurationRoot _cachedConfiguration;
+
+        protected IConfigurationRoot Configuration
+        {
+            get
+            {
+                if (_cachedConfiguration != null)
+                {
+                    return _cachedConfiguration;
+                }
+
+                var builder = new ConfigurationBuilder()
+                    .SetBasePath(Directory.GetCurrentDirectory())
+                    .AddJsonFile("appsettings.json", optional: false, reloadOnChange: true)
+                    .AddJsonFile($"appsettings.{EnvironmentName}.json", optional: true)
+                    .AddEnvironmentVariables();
+
+                _cachedConfiguration = builder.Build();
+                ConfigurationUtils.SetConfiguration(_cachedConfiguration);
+
+                return _cachedConfiguration;
+            }
+        }
+
+        public TestingSettings ODataTestSettings
+        {
+            get { return Configuration.GetSection("SitefinityODataTestSettings").Get<TestingSettings>(); }
+        }
+
+        public string AccessToken { get; set; }
+
+        private string AuthenticateUrl
```

Comment:
Suggestion: Consider using an expression body shortening to: 
```
private string AuthenticateUrl
{
    get => ODataTestSettings.BaseUrl + "/Sitefinity/Authenticate/OpenID/connect/token";
}
```

Suggestion: Move the path to login to a configuration file, or make it an environment variable. That way if the path to the login changes in future versions, we can update it without a deployment.